tests/unit/value_scaling_tests.cpp

```diff
@@ -59,18 +61,22 @@ TEST(ValueScalingTest, arpMidiCvGateValueScaling) {
 	CHECK_EQUAL(50, computeCurrentValueForArpMidiCvGate(INT32_MAX));
 }
 
-TEST(ValueScalingTest, arpMidiCvRatchetValueScaling) {
+TEST(ValueScalingTest, consistentArpAndMenuMaxValues) {
+	CHECK_EQUAL(kMaxMenuValue, kMaxPresetArpRhythm);
```

Comment:
this should probably be a `static_assert` near the definition of `kMaxPresetArpRhythm`

Reply:
On reflection, I think the test needs to be separate:

Adding a static assert next to

```const int32_t kMaxPresetArpRhythm = kMaxMenuValue;```

is just redundant. Anyone tempted to change that assignment will also change the static assert.

The test being lexically distant is an advantage, because it will not be modified at the same time without any consideration.

I'll add some commentary about _why_ they need to be the same, though.

Reply:
I think this has the same issue, it'll just get modified to pass the test. Do any of the other tests fail if this isn't true?

Reply:
I'll test and see. Still, a comment where the values are defined and an explicit test referring to the comment is IMO a reasonable precaution against accidental modification of depenedent parameters.

I mean, they're dependent already. This is just a tiny bit safety for when someone is neat that footgun.

Reply:
I think the test is the way to go: no other test fails, and the build succeeds if I just increase the number of arp rhythms by one, despite that messing up menus.

I can change it to the a static assert instead, but as said I believe being textually distant is an advantage in this case: more chance of actually stopping and thinking instead of working on autopilot.

Just let me know which way you want it.